certification/internal/shell/base_on_ubi_test.go

```diff
@@ -1,41 +1,65 @@
 package shell
 
 import (
+	"strings"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
+	log "github.com/sirupsen/logrus"
 )
 
 var _ = Describe("BaseOnUBI", func() {
 	var (
-		engine      FakePodmanEngine
-		base_on_ubi BaseOnUBICheck
+		baseOnUbiCheck BaseOnUBICheck
+		originalEngine cli.PodmanEngine
+		fakeEngine     FakePodmanEngine
+		lines          []string
 	)
 
 	BeforeEach(func() {
-		engine = FakePodmanEngine{
+		fakeEngine = FakePodmanEngine{
 			RunReportStdout: `ID="rhel"
 NAME="Red Hat Enterprise Linux"
 `,
 			RunReportStderr: "",
 		}
+		lines = strings.Split(fakeEngine.RunReportStdout, "\n")
+		originalEngine = podmanEngine
+	})
+	AfterEach(func() {
+		podmanEngine = originalEngine
 	})
-
 	Describe("Checking for UBI as a base", func() {
 		Context("When it is based on UBI", func() {
+			BeforeEach(func() {
+				podmanEngine = fakeEngine
+			})
 			It("should succeed the check", func() {
-				ok, err := base_on_ubi.validate(engine, "dummy/image")
+				ok, err := baseOnUbiCheck.validate(lines)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ok).To(BeTrue())
+			})
+			It("should pass Validate", func() {
+				ok, err := baseOnUbiCheck.Validate("dummy/image")
```

Comment:
Now I have the peace of mind ;) Thanks!

IMO testing the Validate() method is enough. It will test the validate() automatically and we don't need to maintain implementation in both tests and the method (e.g. line 27: if at some point we decide to split lines based on another char, we have to update the code and the test).

Reply:
I was literally thinking about that as well. Let me address that and add some comments about the fakery and then I'll repush.